### PR TITLE
Creating tests for org user role functionality

### PR DESCRIPTION
**Pull request by rememberlenny**

### CODE REVIEW

File: static_src/test/functional/user_role.spec.js [hunk 1]
```diff
@@ -1,58 +1,150 @@
...
+      });
 
-        expect(true).toBe(true);
+      describe('shouldn\'t have permission to edit fields on org X pages', function () {
+        it('set url', function () {
+          pageUrl = urlOrgX;
+        });
+
+        it('verify cookie', function () {
+          cookieResult = userRoleElement.setAndGetUserRole(browser, pageUrl, cookieValue);
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: userRoleElement already has the `browser` object

File: static_src/test/functional/user_role.spec.js [hunk 2]
```diff
@@ -1,58 +1,150 @@
...
-        cookieResult = browser.getCookie('testing_user_role').value;
-        expect(cookieResult).toBe('org_manager_org_x');
+    describe('As org manager Y', function () {
+      it('set cookie to org Y manager', function () {
+        cookieValue = cookieManagerOrgY;
+      });
 
-        expect(true).toBe(true);
+      describe('shouldn\'t have permission to edit fields on org X pages', function () {
+        it('set url', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: set url to org X

File: static_src/test/functional/user_role.spec.js [hunk 3]
```diff
@@ -1,58 +1,150 @@
...
+          pageUrl = urlOrgX;
+        });
+
+        it('verify cookie', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: more description. something like: verifies that we are currently org manager X in this session

File: static_src/test/functional/user_role.spec.js [hunk 4]
```diff
@@ -1,58 +1,150 @@
...
+          cookieResult = userRoleElement.setAndGetUserRole(browser, pageUrl, cookieValue);
+          expect(cookieResult).toBe(cookieManagerOrgY);
+        });
+
+        it('verify org Y manager can modify org X page', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: `can` -> `cannot`

File: static_src/test/functional/user_role.spec.js [hunk 5]
```diff
@@ -1,58 +1,150 @@
...
+          expect(userRoleElement.isFirstUserRoleEnabled(browser)).toBe(false);
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: same as https://github.com/18F/cg-dashboard/pull/1074/files#r113179630

File: static_src/test/functional/user_role.spec.js [hunk 6]
```diff
@@ -1,58 +1,150 @@
...
+        });
       });
 
-      it('not be able to edit roles for org Y', function () {
-        expect(true).toBe(true);
+      describe('should have permission to edit fields on org Y pages', function () {
+        it('set url', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: same as https://github.com/18F/cg-dashboard/pull/1074/files#r113180194

File: static_src/test/functional/user_role.spec.js [hunk 7]
```diff
@@ -1,58 +1,150 @@
...
-    describe('space manager for org X space TT then they should', function () {
-      it('not be able to edit roles for org X', function () {
-        expect(true).toBe(true);
+    describe('As org manager X', function () {
+      it('set cookie to org X manager', function () {
+        cookieValue = cookieManagerOrgX;
       });
+
+      describe('shouldn\'t have permission to edit fields on org Y pages', function () {
+        it('set url', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: same as https://github.com/18F/cg-dashboard/pull/1074/files#r113180194

File: static_src/test/functional/user_role.spec.js [hunk 8]
```diff
@@ -84,7 +176,6 @@ describe('User roles', function () {
       });
 
       it('not be able to edit roles for space YY', function () {
-        expect(true).toBe(true);
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: we still are adding these right?

Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: maybe we should split it to another PR too

File: static_src/test/functional/user_role.spec.js [hunk 9]
```diff
@@ -1,58 +1,150 @@
...
+        it('set url', function () {
+          pageUrl = urlOrgY;
+        });
+
+        it('verify cookie', function () {
+          cookieResult = userRoleElement.setAndGetUserRole(browser, pageUrl, cookieValue);
+          expect(cookieResult).toBe(cookieManagerOrgX);
+        });
+
+        it('verify org X manager can modify org Y page', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: `can` -> `cannot`

File: static_src/test/functional/user_role.spec.js [hunk 10]
```diff
@@ -1,58 +1,150 @@
...
+        browser.url(urlOrgY);
+
+        browser.waitForExist('.test-users');
+        userRoleElement = new UserRoleElement(browser, browser.element('.test-users') );
+
+        expect(userRoleElement.isVisible()).toBe(true);
+      });
+
+      it('should not be able to edit roles for org Y', function () {
+        expect(userRoleElement.isUserOrgManager(guid_manager_orgX)).toBe(false);
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: `guid_manager_orgX` -> `guidManagerOrgX`

File: static_src/test/functional/user_role.spec.js [hunk 11]
```diff
@@ -1,58 +1,150 @@
...
+        browser.url(urlOrgX);
+
+        browser.waitForExist('.test-users');
+        userRoleElement = new UserRoleElement(browser, browser.element('.test-users') );
+
+        expect(userRoleElement.isVisible()).toBe(true);
+      });
+
+      it('should be able to edit roles for org X', function () {
+        expect(userRoleElement.isUserOrgManager(guid_manager_orgX)).toBe(true);
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: same as https://github.com/18F/cg-dashboard/pull/1074/files#r113188117

File: static_src/test/functional/user_role.spec.js [hunk 12]
```diff
@@ -1,58 +1,150 @@
...
+  const guidManagerOrgX = 'org-manager-x-uid-601d-48c4-9705',
+    guidManagerOrgY = 'org-manager-y-uid-601d-48c4-9705',
+    cookieManagerOrgY = 'org_manager_org_y',
+    cookieManagerOrgX = 'org_manager_org_x',
+    cookieManagerSpaceXX = 'space_manager_space_xx',
+    cookieManagerSpaceYY = 'space_manager_space_yy',
+    urlOrgY = '/#/org/user_role-org_y-ffe7-4aa8-8e85-94768d6bd250',
+    urlOrgX = '/#/org/user_role-org_x-ffe7-4aa8-8e85-94768d6bd250';
 
   describe('User role cookie test', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: is this section needed since all the setCookie and getCookie functionality is now in the userRoleElement?

File: static_src/test/functional/user_role.spec.js [hunk 13]
```diff
@@ -1,59 +1,150 @@
...
+        browser.url(urlOrgX);
+
+        browser.waitForExist('.test-users');
+        userRoleElement = new UserRoleElement(browser, browser.element('.test-users'));
+
+        expect(userRoleElement.isVisible()).toBe(true);
+      });
+
+      it('should be able to edit roles for org X', function () {
+        expect(userRoleElement.isUserOrgManager(guidManagerOrgX)).toBe(true);
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: i'm not sure if these assertions meet the test statement. maybe we should check for if the checkboxes are enabled / disabled

File: static_src/test/functional/user_role.spec.js [hunk 14]
```diff
@@ -1,59 +1,150 @@
...
+      it('should navigates to org Y ', function () {
+        browser.url(urlOrgY);
+
+        browser.waitForExist('.test-users');
+        userRoleElement = new UserRoleElement(browser, browser.element('.test-users'));
+
+        expect(userRoleElement.isVisible()).toBe(true);
+      });
+
+      it('should not be able to edit roles for org Y', function () {
```
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: same as https://github.com/18F/cg-dashboard/pull/1074/files#r113191465

File: static_src/test/functional/user_role.spec.js (hunk 12 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: not sure why this is outdated now

File: static_src/test/functional/user_role.spec.js (hunk 1 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 5 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 10 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 11 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/pageobjects/user_role.element.js [hunk 15]
```diff
@@ -9,5 +9,36 @@ import BaseElement from './base.element';
...
 export default class UserRoleElement extends BaseElement {
+  // TODO: Look into why "this.element('#org_role' + guid).checked
+  // does not work. Currently, it doesn't work that way.
+  setAndGetUserRole(testUrl, cookieValue) {
+    this.setUserRole(testUrl, cookieValue);
+    return this.getUserRole(testUrl);
+  }
 
+  setUserRole(testUrl, cookieValue) {
+    this.browser.url(testUrl);
```
Comment (jcscottiii) [inline static_src/test/functional/pageobjects/user_role.element.js]: navigating to a new page shouldn't be here in the element. (Unless the element actually would navigate to another page in real life)

Comment (rememberlenny) [inline static_src/test/functional/pageobjects/user_role.element.js]: The navigation was previously necessary because the cookie wouldn't set without being on a page. It doesn't *need* to be there, but it helps prevent an issue where a cookie is being set but the browser isnt on a page.

File: static_src/test/functional/user_role.spec.js (hunk 2 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 3 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 4 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 6 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 7 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: fixed

File: static_src/test/functional/user_role.spec.js (hunk 8 above)
Comment (jcscottiii) [inline static_src/test/functional/user_role.spec.js]: will be split into another PR
